abnf/odata-aggregation-testcases.yaml

```diff
@@ -216,6 +244,43 @@ TestCases:
     Rule: queryOptions
     Input: $apply=aggregate(Amount with sum as Total)
 
+  - Name: aggregate - arithmetic expression
+    Rule: queryOptions
+    Input: $apply=aggregate((Amount sub Cost) mul TaxRate with sum as Tax)
+    Expect:
+      - aggregatableFactor:(Amount sub Cost)
+      - aggregatableFactor:TaxRate
+
+  - Name: aggregate - arithmetic expression with literals
+    Rule: queryOptions
+    Input: $apply=aggregate((TaxRate sub 1) mul 'P1D' with average as Stuff)
+
+  - Name: aggregate - forbidden arithmetic on collection
+    Rule: queryOptions
+    FailAt: 30
+    Input: $apply=aggregate(Sales/Amount sub Sales/Cost with sum as TotalAmount)
+
+  - Name: aggregate - allowed arithmetic on collection
+    Rule: queryOptions
+    Input: $apply=addnested(Sales,compute(Amount sub Cost as Profit) as
+      Profits)/aggregate(Profits/Profit with sum as TotalAmount)
+    Expect:
+      - aggPathPrimitive:Profits/Profit
+
+  - Name: aggregate - forbidden arithmetic
+    Rule: queryOptions
+    FailAt: 37
+    Input: $apply=aggregate(Amount sub Discounts with sum as TotalAmount)
+
+  - Name: aggregate - functional expression
+    Rule: queryOptions
+    Input: $apply=aggregate(Self.sqrt(Number=Amount mul Amount sub Cost mul Cost) with sum as TotalAmount)
+
+  - Name: aggregate - functional expression on entity forbidden
```

Comment:
TC 2022-04-14 agrees to allow functional expressions on `firstMemberExpr`s.

Reply:
`firstMemberExpr` allows too much (for example, key predicates), let's allow `grpPropPath` instead.

Reply:
TC 2022-04-28: Change prefixes of data aggregation path rules
* `grp` -> `single`
* `agg` -> `any`
* `nst` -> `nest`